test/integration/test_pkg_signing.py

```diff
@@ -54,3 +55,105 @@ def verify(ref, artifacts_folder, signature_folder, files):
     assert "Verifying ref:  pkg/0.1" in c.out
     assert "VERIFYING  conanfile.py" not in c.out  # It doesn't re-verify previous contents
     assert "VERIFYING  conan_sources.tgz" in c.out
+
+
+def test_pkg_sign_check_integrity():
+    c = TestClient(default_server_user=True)
+    c.save({"conanfile.py": GenConanfile("pkg", "0.1").with_exports("export/*")
+            .with_exports_sources("export_sources/*").with_package_file("myfile", "mycontents!"),
+            "export/file1.txt": "file1!",
+            "export_sources/file2.txt": "file2!"})
+    signer = textwrap.dedent(r"""
+        import os
+        from conan.api.output import ConanOutput
+        from conan.api.model.refs import PkgReference
+        from conan.errors import ConanException
+
+
+        def sign(ref, artifacts_folder, signature_folder, output):
+            output.info("Signing reference")
+            output.info(f"Signing folder: {artifacts_folder}")
+            files = []
+            for f in sorted(os.listdir(artifacts_folder)):
+                if os.path.isfile(os.path.join(artifacts_folder, f)):
+                    files.append(f)
+            output.info(f"Signing files: {sorted(files)}")
+            signature = os.path.join(signature_folder, "signature.asc")
+            open(signature, "w").write("\n".join(files))
+            output.info("Package signature creation: ok")
+
+        def verify(ref, artifacts_folder, signature_folder, files, output):
+            output.info("Verifying reference")
+            output.info(f"Verifying folder: {artifacts_folder}")
+            signature = os.path.join(signature_folder, "signature.asc")
+
+            # This is to check if the package was signed or not
+            if isinstance(ref, PkgReference):
+                download_file_path = os.path.join(artifacts_folder, "conan_package.tgz")
+            else:
+                download_file_path = os.path.join(artifacts_folder, "conanfile.py")
+
+            if not os.path.isfile(download_file_path) and not os.path.isfile(signature):
+                output.warning("Could not verify unsigned package")
+                return
```

Comment:
This is maybe the last part to improve, as this is the way I found to check if the package was signed or not (it was built but never "uploaded")

Reply:
Yes, it seems the system needs to support the handling and reporting of packages not signed, probably with some opt-in/opt-out logic to error or not